lib/msf/core/modules/external/shim.rb

```diff
@@ -4,7 +4,7 @@
 class Msf::Modules::External::Shim
   def self.generate(module_path, framework)
     mod = Msf::Modules::External.new(module_path, framework: framework)
-    return '' unless mod.meta
+    return nil unless mod.meta
```

Comment:
So this will raise an error for every module that the current environment cannot run, like if you don't have a version of Python. This can get kinda noisy, which is why an empty module used to be returned. Maybe we should do some logging here instead of returning `nil`?

Reply:
That was the point. Silently failing to load a module is worse for someone trying to debug a module than logging _something_. It logs to the .msf4/ logs/framework.log file, so it's not right in front of the user, but there's enough info that a developer has a fighting chance of debugging in the field.

Reply:
Ideally, all of our users eventually get all of the dependencies. That's our plan anyway, or they're going to miss out on more modules over time. We should tolerate half-configured environments, but not make it impossible to tell if they are.

Reply:
Maybe as a followup, I could look at stderr for each failure, and only emit unique failures once. That makes the lack of python a single message, rather than dozens.

Reply:
Or maybe just one log line per module instead of a whole stacktrace unless debug is enabled.

Reply:
Yeah, agree we don't need a full stracktrace.